source/server/filter_chain_manager_impl.h

```diff
@@ -128,9 +154,79 @@ class FilterChainManagerImpl : public Network::FilterChainManager,
   findFilterChainForSourceIpAndPort(const SourceIPsTrie& source_ips_trie,
                                     const Network::ConnectionSocket& socket) const;
 
-  // Mapping of FilterChain's configured destination ports, IPs, server names, transport protocols
-  // and application protocols, using structures defined above.
-  DestinationPortsMap destination_ports_map_;
+  struct FilterChainLookup {
```

Comment:
I'm not sure I get here what a `FilterChainLookup` is; can you write a detailed block comment?